### DISCUSSION TITLE
Verified commits on github.dev web editor (not codespaces)

### DISCUSSION THREAD

**Discussion by Katsute** [community/community#5841] [Codespaces] [answered]
The github.dev `.` editor should have verified commits just like the current web editor.

https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-codespaces ← this solution is not applicable for people who are not using codespaces, we don't have a codespaces tab.

**Comment by bamurtaugh** [comment, marked as answer]
Thanks for opening @Katsute! We've noted this as a feature request as folks have expressed interest in this area.

**Reply by joyceerhl** [reply]
You can try out our support for verified commits by setting `"remoteHub.experimental.useNewCommitApi": true` in your VS Code settings.json file.

**Reply by Anselmoo** [reply]
Is that also working in combination with locally running VSCode connected to CodeSpaces?

**Reply by Anselmoo** [reply]
@Katsute @joyceerhl is there a general possibility to change the signing to `GitHub-Username` to `personal-Username`? 
_I am a little bit afraid to break the container settings by manually overwriting the `git-configs`_

**Reply by joyceerhl** [reply]
@Anselmoo When running in Codespaces, you will need to manually configure commit signing following https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-codespaces. 

If you would like the commit to be associated with a different GitHub username, you can do this by making sure you are signed into the relevant GitHub account when accessing github.dev. There is no other option to change the username because GitHub's APIs do not currently support commit signing together with username configuration: https://docs.github.com/en/graphql/reference/mutations#authorship. You _can_ set the email associated with github.dev commits by changing your primary email in your GitHub settings.

Note also that commit signing in the serverless github.dev editor is now available by default without needing the `"remoteHub.experimental.useNewCommitApi": true` setting mentioned above.

**Reply by Anselmoo** [reply]
@joyceerhl thx a lot of the very detailing descriptions, which work very nicely. 

And I didn't know that `Codespaces` offers so much fine-tuning. My favorite is the automatic installation of `.dotfiles`

thx again
